src/hooks/useCameraSelect.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import { toast } from "sonner";
+
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+import usePreferredMediaDevice from "@/hooks/usePreferredMediaDevice";
+
+interface UseCameraSelectOptions {
+  onChange?: (deviceId: string) => void;
+}
+
+const useCameraSelect = ({ onChange }: UseCameraSelectOptions = {}) => {
+  const [devices, setDevices] = useState<any[]>([]);
+  const [selectedDeviceId, setSelectedDeviceId] = useState<string>("");
+  const { preferredDeviceId, setDeviceId } = usePreferredMediaDevice();
+
+  useEffect(() => {
+    const getDevices = async () => {
+      try {
+        const mediaDevices = await navigator.mediaDevices.enumerateDevices();
+        const videoDevices = mediaDevices.filter(
+          ({ kind }) => kind === "videoinput",
+        );
+        setDevices(videoDevices);
+
+        let initialId = preferredDeviceId;
```

Comment:
naming here isn't great (initialId and exists). Change it to something more appropriate like deviceId and filteredId.